### PR TITLE
Migrate `queries/vulnerabilities.ts` to use `@app/client`

### PR DESCRIPTION
**Pull request by chirino**
Added a convertQuery function that converts the results of the queries to that expected type by the pages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,5 +69,6 @@
     "ts-node": "^10.9.1",
     "type-fest": "^3.13.0",
     "typescript": "^5.5.3"
-  }
+  },
+  "packageManager": "pnpm@9.6.0+sha512.38dc6fba8dba35b39340b9700112c2fe1e12f10b17134715a4aa98ccf7bb035e76fd981cf0bb384dfa98f8d6af5481c2bef2f4266a24bfa20c34eb7147ce0b5e"
```
Comment (carlosthe19916) [inline package.json:73]: Just out of curiosity, what does this buy us? I've used pnpm in the past but I admit I never used it like this.

**Review by carlosthe19916 — APPROVED** [review]
@chirino Sorry, I needed to push the "update button" in your PR to synchronize it and make the recent "format check" script to run. 

LGTM! Feel free to push the merge button whenever you like.

**Comment by chirino** [comment]
Sorry thats me using pnpm from muscle memory. I think corepack adds that so that the same pnpm version can be reused in subsequent builds. While I think pnpm lock files are one of the nicest, we likely won't be using it in production builds so I'll remove that to avoid noise.

File: package.json (hunk 1 above)
Comment (chirino) [reply package.json]: removed /w rebase.

**Pull request merged**